stl/src/vector_algorithms.cpp

```diff
@@ -433,6 +439,625 @@ __declspec(noalias) void __cdecl __std_reverse_copy_trivially_copyable_8(
+                        }
+                    }
+                    // Horizontal part done, results are saved, now need to see if there is another portion to process
+
+                    if constexpr (_Traits::_Has_portion_max) {
+                        // Either the last portion or wrapping point reached, need to determine
+                        _Portion_size = _Byte_length(_First, _Last) & ~size_t{0xF};
+                        if (_Portion_size == 0) {
+                            break; // That was the last portion
+                        }
+                        // Start next portion to handle the wrapping indices. Assume _Cur_idx is zero
+                        constexpr size_t _Max_portion_size = _Traits::_Portion_max * 16;
+                        if (_Portion_size > _Max_portion_size) {
+                            _Portion_size = _Max_portion_size;
+                        }
+
+                        _Advance_bytes(_Stop_at, _Portion_size);
+                        // Indices will be relative to the new base
+                        _Base = static_cast<const char*>(_First);
+                        // Load values and if unsigned adjust them to be signed (for signed vector comparisons)
+                        _Cur_vals =
+                            _Traits::_Sign_cor(_mm_loadu_si128(reinterpret_cast<const __m128i*>(_First)), _Sign);
+
+                        if constexpr ((_Mode & _Mode_min) != 0) {
+                            _Cur_vals_min = _Cur_vals;
+                            _Cur_idx_min  = _mm_setzero_si128();
+                        }
+                        if constexpr ((_Mode & _Mode_max) != 0) {
+                            _Cur_vals_max = _Cur_vals;
+                            _Cur_idx_max  = _mm_setzero_si128();
+                        }
+
+                        continue;
+                    } else {
+                        break; // No wrapping, so it was the only portion
+                    }
+                }
+                // This is the main part, finding vertical minimum/maximum
+
+                // Load values and if unsigned adjust them to be signed (for signed vector comparisons)
+                _Cur_vals = _Traits::_Sign_cor(_mm_loadu_si128(reinterpret_cast<const __m128i*>(_First)), _Sign);
+
+                if constexpr ((_Mode & _Mode_min) != 0) {
+                    // Looking for the first occurence of minimum, don't overwrite with newly found occurences
+                    const __m128i _Is_less = _Traits::_Cmp_gt(_Cur_vals_min, _Cur_vals); // _Cur_vals < _Cur_vals_min
+                    _Cur_idx_min = _mm_blendv_epi8(_Cur_idx_min, _Cur_idx, _Is_less); // Remember their vertical indices
+                    _Cur_vals_min = _Traits::_Min(_Cur_vals_min, _Cur_vals, _Is_less); // Update the current minimum
+                }
+
+                if constexpr (_Mode == _Mode_max != 0) {

```

Comment:
the != 0 is not needed for this one (I hope 😬), also the precedence is confusing